qubes_config/policy_editor/policy_editor.py

```diff
@@ -0,0 +1,537 @@
+            show_error(self.main_window, "File already exists",
+                       f"Policy file: {new_name} already exists.")
+            return
+
+        self.token = "any"
+        self._set_policy_file(new_name, "")
+
+
+    def _save(self, *_args):
+        """Save changes. If successful, return True."""
+        try:
+            self.policy_client.policy_replace(self.filename,
+                                              self.policy_text, self.token)
+        except subprocess.CalledProcessError as ex:
+            err_msg = "An error occurred while trying to save the policy" \
+                      " file:\n"
+            if ex.stdout:
+                err_msg += ex.stdout.decode()
+            else:
+                err_msg += str(ex)
+            show_error(self.main_window, "Failed to save policy", err_msg)
+            return False
+        self.open_policy_file(self.filename)
+        self.source_buffer.set_modified(False)
+        self.action_items['save'].set_enabled(False)
+        self.action_items['save_exit'].set_enabled(False)
+        return True
+
+    def _save_exit(self, *_args):
+        if self._save():
+            self._quit()
+
+    def _open(self, *_args):
+        self.file_select_handler.show_dialog()
+
+    def _redo(self, *_args):
+        self.source_buffer.redo()
+
+    def _undo(self, *_args):
+        self.source_buffer.undo()
+
+    def _copy(self, *_args):
+        self.source_buffer.copy_clipboard(self.clipboard)
+
+    def _paste(self, *_args):
+        self.source_buffer.paste_clipboard(self.clipboard, None, True)
+
+    def _about(self, *_args):
+        self.about_window.show()
+
+    def _toggle_help(self, *_args):
+        self.help_window.set_visible(not self.help_window.get_visible())
+
+    def _set_policy_file(self, name, contents):
+        self.filename = name
+        self.source_buffer.set_text(contents)
+        self.window_title = 'Qubes OS Policy Editor - ' + self.filename
+        self.main_window.set_title(self.window_title)
+        self.source_buffer.set_modified(False)
+        self.action_items['save'].set_enabled(False)
+        self.action_items['save_exit'].set_enabled(False)
+
+    def open_policy_file(self, name: Optional[str]):
+        if name is None:
+            return
+        self.source_view.set_sensitive(True)
+        self.error_info.set_visible(True)
+        if name == '':
+            text = "Create new file or open an existing one."
+            self.token = None
+            self.source_view.set_sensitive(False)
+            self.error_info.set_visible(False)
+        else:
+            try:
+                text, self.token = self.policy_client.policy_get(name)
+            except subprocess.CalledProcessError:
+                response = ask_question(
+                    self.main_window, "Policy file not found",
+                    "File {} not found. Do you want to create a "
+                    "new policy file?".format(name))
```

Comment:
It could also mean no access to the file, I think you'll get then return code 126 (you have it as an attribute on the exception), but it's probably nicer to wrap it in a different exception in admin_client.